### DISCUSSION TITLE
content-type-parser:run preventing jest from exiting

### DISCUSSION THREAD

**Discussion by smallandsimple** [fastify/fastify#4174] [Q&A] [answered]
Hello, I am having an issue I could use some help with.

Using: JEST
Test:
```
test('POST `/auth/valid` route - with valid credentials', async () => {
    const { body } = await fastify.inject()
        .post('/auth/valid')
        .body({ username: 'test', password: 'test' })
        .end()
    const response = JSON.parse(body)
    expect(response).toBe(true)
})
```

**The test is successful, however, I get the following message from JEST**:
`Jest has detected the following 4 open handles potentially keeping Jest from exiting: content-type-parser:run`

In my `afterAll` I am calling `await fastify.close()`.

Any ideas?

**Comment by Eomm** [comment, marked as answer]
Could you add a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)?

Without it, we are unable to help you.

**Reply by smallandsimple** [reply]
Well, it just so happened that when I made a code change that was completely unrelated to the error today, somehow it resolved it.  I can only assume that something in the response data being sent back by one of the controllers was the issue. (shrug)

Thank you anyway!
